lib/matplotlib/axes/_base.py

```diff
@@ -4162,7 +4162,18 @@ def get_default_bbox_extra_artists(self):
         Artists are excluded either by not being visible or
         ``artist.set_in_layout(False)``.
         """
-        return [artist for artist in self.get_children()
+
+        artists = self.get_children()
+        if not (self.axison and self._frameon):
+            # don't do bbox on spines if frame not on.
+            for spine in self.spines.values():
+                artists.remove(spine)
+
+        if not self.axison:
+            for _axis in self._get_axis_list():
+                artists.remove(_axis)
```

Comment:
Just wondering out loud... would it be more efficient to build the list of artists with a complicated if-logic, or remove elements from a list based on some if-conditions?

Reply:
Not quite sure what you are asking for here; making the if-statement below much larger?  Maybe that'd be a bit faster, but a lot harder to follow.  This part of the code doesn't seem like a bottleneck to me, but I upgrade my computer often :wink:

Reply:
fwiw this suggests that we should just get rid of axison and frameon and instead directly manipulate the artists' visibility, see discussion at https://github.com/matplotlib/matplotlib/pull/11691, etc.

Reply:
Running the following w/ `on=True` and `on=False` yileds 49.15s and 49.25s respectively on my machine, which considering I didn't stop doing other work says to me the difference is in the noise...

```python

import matplotlib.pyplot as plt
import numpy as np
import time

def testSpeed(on=False):
    fig, ax = plt.subplots()
    for a in range(5000):
        ax.scatter(np.random.rand(200), np.random.rand(200))

    if not on:
        ax.axison = False
        ax._frameon = False
    art = ax.get_default_bbox_extra_artists()

# at the beginning:
start_time = time.time()
for i in range(3):
    print(i)
    testSpeed(on=True)
print(time.time()-start_time)
```

On the opposite end of the spectrum:

```python
def testSpeed(on=False):
    fig, ax = plt.subplots()
    for a in range(1):
        ax.scatter(np.random.rand(200), np.random.rand(200))

    if not on:
        ax.axison = False
        ax._frameon = False
    art = ax.get_default_bbox_extra_artists()

# at the beginning:
start_time = time.time()
for i in range(300):
    print(i)
    testSpeed(on=False)
print(time.time()-start_time)
```

I got True: 6.4s, False: 6.22s  

So I don't think this adds a drag on the system unless those tests are wrong somehow...

Reply:
(OFF TOPIC: I agree that would make the logic in the "if" statement here a lot easier.  For a user, however, its hard for them to know how to manipulate the underlying objects - if they want to make an axes invisble, they are likley to look at the axes artist methods and docs and be frustrated because they don't see a method to toggle the spine visibility, and not know they need to manipulate the axis artists attached to that axes.  I have zero problem with making `axison` a thin wrapper around setting the visibility of the underlying spine objects, but if we are to deprecate `axison` we need a clear way forward to make sure users know how to do the equivalent operation.  )

Reply:
I agree we could do a better job at documenting sub-artists...

Reply:
@WeatherGod are you still wanting to block on this?  I think performance wise its OK